test-cloud/test-cloud-remote/src/test/java/org/kie/cloud/integrationtests/sso/AuthSSOIntegrationTest.java

```diff
@@ -0,0 +1,178 @@
+    public void testMultipleDifferentProcessesOnSameKieServer() {
+        KieServerInfo serverInfo = kieServerClient.getServerInfo().getResult();
+        WorkbenchUtils.saveContainerSpec(kieControllerClient, serverInfo.getServerId(), serverInfo.getName(), CONTAINER_ID, CONTAINER_ALIAS, PROJECT_GROUP_ID, DEFINITION_PROJECT_NAME, DEFINITION_PROJECT_VERSION, KieContainerStatus.STARTED);
+
+        KieServerClientProvider.waitForContainerStart(deploymentScenario.getKieServerDeployment(), CONTAINER_ID);
+
+        Long userTaskPid = processClient.startProcess(CONTAINER_ID, USERTASK_PROCESS_ID);
+        assertThat(userTaskPid).isNotNull();
+        Long signalTaskPid = processClient.startProcess(CONTAINER_ID, SIGNALTASK_PROCESS_ID);
+        assertThat(signalTaskPid).isNotNull();
+
+        // START of System.out
+        System.out.println(processClient.findProcessInstances(CONTAINER_ID, 0, 10));
+        System.out.println(processClient.getProcessInstance(CONTAINER_ID, userTaskPid).getActiveUserTasks().getItems());
+
+        System.out.println(taskClient.getTaskInstance(CONTAINER_ID, Long.valueOf(1))); // as yoda should be returned not empty list
+
+        ProcessInstance pi = queryClient.findProcessInstanceById(new Long(1));
+        System.out.println(pi);
+        System.out.println(pi.getInitiator()); //Initiator is user id of Yoda
+        System.out.println(Arrays.toString(pi.getActiveUserTasks().getTasks()));
+        System.out.println(pi.getActiveUserTasks().getTasks()[0].getActualOwner());
+        System.out.println(pi.getActiveUserTasks().getTasks()[0].getCreatedBy());
+        System.out.println(pi.getActiveUserTasks().getTasks()[0].getStatus());
+
+        System.out.println(taskClient.findTasksAssignedAsPotentialOwner(USER_YODA, 0, 10)); //task is assigned to Yoda, but list is empty
+        // END of System.out
+
+        finishUserTaskProcess();
+        finishSignalTaskProcess(signalTaskPid);
+
+        ProcessInstance userTaskPi = processClient.getProcessInstance(CONTAINER_ID, userTaskPid);
+        assertThat(userTaskPi).isNotNull();
+        assertThat(userTaskPi.getState()).isEqualTo(org.kie.api.runtime.process.ProcessInstance.STATE_COMPLETED);
+
+        ProcessInstance signalTaskPi = processClient.getProcessInstance(CONTAINER_ID, signalTaskPid);
+        assertThat(signalTaskPi).isNotNull();
+        assertThat(signalTaskPi.getState()).isEqualTo(org.kie.api.runtime.process.ProcessInstance.STATE_COMPLETED);
+    }
+
+    private void finishUserTaskProcess() {
+        List<TaskSummary> tasks = taskClient.findTasks(USER_YODA, 0, 10);
+        assertThat(tasks).isNotNull().hasSize(1);
+
+        taskClient.completeAutoProgress(CONTAINER_ID, tasks.get(0).getId(), USER_YODA, null);
+    }
+
+    private void finishSignalTaskProcess(Long pid) {
+        List<String> signals = processClient.getAvailableSignals(CONTAINER_ID, pid);
+        assertThat(signals).isNotNull().hasSize(1);
```

Comment:
Simplify these 2 assertions to 
```
        assertThat(signals)
                .hasSize(1)
                .contains(SIGNAL_NAME, atIndex(0));
```
And be aware of this pattern in general. The code will be more concise (no need for separate size checkin / getting at given index and checking for equality) more readable and you'll get better error messages.